kfs-core/src/main/java/edu/arizona/kfs/fp/document/web/struts/ProcurementCardAction.java

```diff
@@ -0,0 +1,125 @@
+                String introNoteMessage = getKualiConfigurationService().getPropertyValueAsString(KFSKeyConstants.MESSAGE_RETURN_NOTE_TEXT_INTRO) + KRADConstants.BLANK_SPACE;
+
+                // build out full message
+                disapprovalNoteText = introNoteMessage + reason;
+                int disapprovalNoteTextLength = disapprovalNoteText.length();
+
+                // get note text max length from DD
+                int noteTextMaxLength = getDataDictionaryService().getAttributeMaxLength(Note.class, KRADConstants.NOTE_TEXT_PROPERTY_NAME).intValue();
+
+                if (StringUtils.isBlank(reason) || (disapprovalNoteTextLength > noteTextMaxLength)) {
+                    // figure out exact number of characters that the user can enter
+                    int reasonLimit = noteTextMaxLength - disapprovalNoteTextLength;
+
+                    if (reason == null) {
+                        // prevent a NPE by setting the reason to a blank string
+                        reason = KFSConstants.EMPTY_STRING;
+                    }
+                    return this.performQuestionWithInputAgainBecauseOfErrors(mapping, form, request, response, KRADConstants.DOCUMENT_DISAPPROVE_QUESTION, getKualiConfigurationService().getPropertyValueAsString(KFSKeyConstants.QUESTION_RETURN_DOCUMENT), KRADConstants.CONFIRMATION_QUESTION, KRADConstants.MAPPING_DISAPPROVE, "", reason, KFSKeyConstants.ERROR_DOCUMENT_RETURN_REASON_REQUIRED, KRADConstants.QUESTION_REASON_ATTRIBUTE_NAME, new Integer(reasonLimit).toString());
+                }
+
+                if (KRADUtils.containsSensitiveDataPatternMatch(disapprovalNoteText)) {
+                    return this.performQuestionWithInputAgainBecauseOfErrors(mapping, form, request, response,
+                            KRADConstants.DOCUMENT_DISAPPROVE_QUESTION, getKualiConfigurationService().getPropertyValueAsString(KFSKeyConstants.QUESTION_RETURN_DOCUMENT),
+                            KRADConstants.CONFIRMATION_QUESTION, KRADConstants.MAPPING_DISAPPROVE, "", reason, RiceKeyConstants.ERROR_DOCUMENT_FIELD_CONTAINS_POSSIBLE_SENSITIVE_DATA,
+                            KRADConstants.QUESTION_REASON_ATTRIBUTE_NAME, REASON);
+                }
+            }
+        }
+        
+        Note returnNote = new Note();
+        returnNote.setNoteText(disapprovalNoteText);
+        NoteService noteService = SpringContext.getBean(NoteService.class);
+        ProcurementCardForm procurementCardForm = (ProcurementCardForm) form;
+        
+        String systemUserPrincipalId = getSystemUserPrincipalId();
+        returnNote = noteService.createNote(returnNote, procurementCardForm.getDocument().getNoteTarget(), systemUserPrincipalId);
+        returnNote.setNotePostedTimestampToCurrent();
+        noteService.save(returnNote);
+        
+        List<RouteNodeInstance> routeNodeInstances = procurementCardForm.getProcurementCardDocument().getDocumentHeader().getWorkflowDocument().getCurrentRouteNodeInstances();
```

Comment:
The return code duplicates exactly what happens in the batch job (only there, the system user is hard-coded). 
This should grant adding a method to the ProcurementCardService to requeueDocument that would be used both in the batch job and in the ProcurementCardAction